openapi-cli/src/main/java/io/ballerina/openapi/generators/schema/ballerinatypegenerators/ArrayTypeGenerator.java

```diff
@@ -66,24 +71,30 @@ public TypeDescriptorNode generateTypeDescriptorNode() throws BallerinaOpenApiEx
         assert schema instanceof ArraySchema;
         ArraySchema arraySchema = (ArraySchema) schema;
         TypeGenerator typeGenerator = TypeGeneratorUtils.getTypeGenerator(arraySchema.getItems());
-        String arrayType = typeGenerator.generateTypeDescriptorNode().toString().trim();
+        TypeDescriptorNode arrayTypeDescriptor = typeGenerator.generateTypeDescriptorNode();
         if (typeGenerator instanceof UnionTypeGenerator) {
-            arrayType = OPEN_PAREN_TOKEN.stringValue() + arrayType + CLOSE_PAREN_TOKEN.stringValue();
+            arrayTypeDescriptor = createParenthesisedTypeDescriptorNode(
+                    createToken(OPEN_PAREN_TOKEN), arrayTypeDescriptor, createToken(CLOSE_PAREN_TOKEN));
         }
-        if (arrayType.endsWith(NILLABLE)) {
-            arrayType = arrayType.substring(0, arrayType.length() - 1);
+        if (arrayTypeDescriptor instanceof OptionalTypeDescriptorNode) {
+            Node node = ((OptionalTypeDescriptorNode) arrayTypeDescriptor).typeDescriptor();
+            arrayTypeDescriptor = ((OptionalTypeDescriptorNode) arrayTypeDescriptor).modify(
+                    node, createIdentifierToken(""));
```

Comment:
Can't we use `minutiaeList` to create white space?

Reply:
Here it should be a token, not a `MinutiaeList`.

Reply:
Why do we need to modify the `arrayTypeDescriptor` ? 🤔

Reply:
This is a special scenario where `--nullable` is given and we create every type descriptor as `nillable`. To avoid duplicate nillable configuration we convert child optional type descriptor nodes to not-optional and make the entire `arrayTypeDescriptor` nillable.

Reply:
Ah, I got it, can't we do early return for only optional arrays without modifying, this is just a suggestions 😃 ? WDYT?

Reply:
This seems to be the more practical approach with our current design. Array descriptor  can be a primitive type, reference or a record.

Reply:
Yeah sure, I suggested by thinking this appending empty space will lead to having formatting issues. no issue if this is an optimal way we can do it. let's go ahead with this. 😊

Reply:
Agreed. Its better to implement the correct node if possible. Updated the implementation as below
```java
if (typeDescriptorNode instanceof OptionalTypeDescriptorNode) {
            Node node = ((OptionalTypeDescriptorNode) typeDescriptorNode).typeDescriptor();
            typeDescriptorNode = (TypeDescriptorNode) node;
        }
```
Please review

Reply:
LGTM 🙂